storm-core/src/jvm/org/apache/storm/scheduler/utils/ArtifactoryConfigLoader.java

```diff
@@ -0,0 +1,388 @@
+     *
+     * resource.aware.scheduler.user.pools.loader.params can be set to any of the following
+     *
+     *  {"artifactory.config.loader.uri": "http://artifactory.example.org:9989/artifactory/confs/my_cluster/mt_user_pool"}
+     *
+     *  {"artifactory.config.loader.uri": "file:///confs/my_cluster/mt_user_pool"}
+     *
+     *  {"artifactory.config.loader.uri": "file:///confs/my_cluster/mt_user_pool", "artifactory.config.loader.timeout.secs" : "60"}
+     **/
+    protected static final String ARTIFACTORY_URI = "artifactory.config.loader.uri";
+    protected static final String ARTIFACTORY_TIMEOUT_SECS="artifactory.config.loader.timeout.secs";
+    protected static final String ARTIFACTORY_POLL_TIME_SECS="artifactory.config.loader.polltime.secs";
+    protected static final String ARTIFACTORY_SCHEME="artifactory.config.loader.scheme";
+    protected static final String ARTIFACTORY_BASE_DIRECTORY="artifactory.config.loader.base.directory";
+    protected static final String LOCAL_ARTIFACT_DIR="scheduler_artifacts";
+    static final String cacheFilename = "latest.yaml";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ArtifactoryConfigLoader.class);
+
+    @SuppressWarnings("rawtypes")
+    private Map _conf;
+    private int _artifactoryPollTimeSecs = 600;
+    private boolean _cacheInitialized = false;
+    // Location of the file in the artifactory archive.  Also used to name file in cache.
+    private String _localCacheDir;
+    private String _artifactoryScheme = "http";
+    private String _baseDirectory = "/artifactory";
+    private int _lastReturnedTime = 0;
+    private int _timeoutSeconds = 10;
+    private Map _lastReturnedValue;
+
+    private static class OurResponseHandler implements ResponseHandler<String> {
+        private static OurResponseHandler singleton = null;
+
+        public static OurResponseHandler getInstance() {
+            if (singleton == null) {
+                singleton = new OurResponseHandler();
+            }
+            return singleton;
+        }
+
+        @Override
+        public String handleResponse(final HttpResponse response) throws IOException {
+            int status = response.getStatusLine().getStatusCode();
+            if (status >= 200 && status < 300) {
+                HttpEntity entity = response.getEntity();
+                return entity != null ? EntityUtils.toString(entity) : null;
+            } else {
+                LOG.error("Got unexpected response code {}", status);
+                return null;
+            }
+        }
+    };
+
+    // Protected so we can override this in testing
+    protected String doGet(String api, String artifact, String host, Integer port) {
+        URIBuilder builder = new URIBuilder().setScheme(_artifactoryScheme).setHost(host).setPort(port);
+
+        if (api != null) {
+            builder.setPath(_baseDirectory + api + artifact);
+        } else {
+            builder.setPath(_baseDirectory + artifact);
```

Comment:
If we filter out double backslashes we can also add in extra ones to be sure we have a slash between api and artifact instead of just assuming there is one.